### ISSUE TITLE
php-fpm service fails with "ERROR: [/etc/php-fpm.conf:117] unknown entry 'systemd_interval'"

### ISSUE THREAD

**Issue by bhamilton** [mayflower/puppet-php#101]
The php-fpm service fails to start on CentOS 6.6 with the following error message:

```
Error: Could not start Service[php-fpm]: Execution of '/sbin/service php-fpm start' returned 1: Starting php-fpm: [21-Jul-2015 02:22:56] ERROR: [/etc/php-fpm.conf:118] unknown entry 'systemd_interval'
==> default: [21-Jul-2015 02:22:56] ERROR: failed to load configuration file '/etc/php-fpm.conf'
==> default: [21-Jul-2015 02:22:56] ERROR: FPM initialization failed
==> default: [FAILED]
```

The issue comes from the file `/etc/init/php-fpm.conf` which contains the following stanza:

```
; When FPM is build with systemd integration, specify the interval,
; in second, between health report notification to systemd.
; Set to 0 to disable.
; Available Units: s(econds), m(inutes), h(ours)
; Default Unit: seconds
; Default value: 10
systemd_interval = 10
```

The above file resource is defined by the class `php::fpm::config` implementing a ERB template. However, the template assumes that the system executing this PHP module is sufficiently recent to support systemd which for CentOS has not been implemented until version 7.0.

A possible solution could pass a boolean checking for the facter value `@operatingsystemrelease` or possibly a combination of the release value and `@osfamily`. I noted that you are already contexually setting pid settings based on `@osfamily` , so this may fit your established conventions.

Manually commenting out this entry appears to resolve the issue.

**Comment by bhamilton** [comment]
Digging in deeper to the as yet unreleased commits in the master branch, it appears https://github.com/mayflower/puppet-php/commit/650f6664686cbfbd90090c31fbe526571d2c5526 actually addresses this issue. Sorry about the noise.

**Comment by globin** [comment]
This is released in 3.3.1 AFAICS
